### PR TITLE
MC-41359 Fix incorrect setting of the SameSite cookie param

### PR DESCRIPTION
**Pull request by ihor-sviziev**
### Description
Port changes from https://github.com/magento/magento2/pull/32462 that fixes https://github.com/magento/magento2/issues/26377 ( MC-41359 ) for Magento 2.3.6-p1 and 2.4.2

Please note that for Magento 2.3.6-p1, we're not patching the file Magento_PageCache/js/form-key-provider.js as it didn't exist yet.

It was added in the scope of https://github.com/magento/magento2/commit/6f68dc560f4318efec57430fe3903dd308ac83d9#diff-fb691f78043f3730b50e6387fc2b9b250a97a808ec7e5412ec404dec5587abd1R303.

The original version was using `$.mage.cookies` that was already patched in this PR scope.
https://github.com/magento/magento2/blob/6f68dc560f4318efec57430fe3903dd308ac83d9/app/code/Magento/PageCache/view/frontend/web/js/page-cache.js#L124

### Fixed Issues (if relevant)

1. <jira_issue_link> <jira_issue_title>
2. ...

### CODE REVIEW

**Comment by ihor-sviziev** [comment]
@viktym could you review?

File: patches.json [hunk 1]
```diff
@@ -1677,5 +1677,17 @@
         }
       }
     }
+  },
+  "MC-41359": {
+    "magento/magento2-base": {
+      "Fixes the issue where SameSite cookie param was set incorrectly.": {
+        ">=2.3.6 <2.3.7": {
```
Comment (hostep) [inline patches.json:1684]: @ihor-sviziev: Pretty sure this needs to be `2.3.6-p1` instead of `2.3.6`
The 2.4.x versions seem to be correct!

Comment (ihor-sviziev) [reply patches.json]: Fixed

**Comment by ihor-sviziev** [comment]
Hi @viktym,

Unfortunately, test results not clear at all. Also, I didn't find any source code for the tests. Could you check if there any issue with my changes?

**Comment by viktym** [comment]
Hi @ihor-sviziev 
Thanks for your contribution!
I'll take a look and give you feedback soon

**Comment by viktym** [comment]
@magento import pr to https://github.com/magento-commerce/quality-patches

**Comment by magento-engcom-team** [comment]
@viktym the pull request successfully imported.

**Comment by viktym** [comment]
@ihor-sviziev tests were passed on internal PR. On your build, for some reason, your patch was trying to apply on Magento 2.3.6. Maybe the configuration in patches.json was ">=2.3.6 <2.3.7"
Please add short Steps to reproduce, Actual and Expected results for patch documentation.
See https://support.magento.com/hc/en-us/sections/360010506631-Patches-available-in-MQP-tool- for reference.

**Comment by ihor-sviziev** [comment]
@viktym, here are few examples of how to reproduce an issue: https://github.com/magento/magento2/pull/32462#issuecomment-799480017

**Comment by viktym** [comment]
Internal PR was merged into the master branch. I'm closing this one.

**Comment by ihor-sviziev** [comment]
Just for reference - changes from this PR were merged in https://github.com/magento/quality-patches/commit/ab51bc42303ea36803d75ac69c7c16fcf3a540d0

**Pull request closed without merging**